internal/generator/generator.go

```diff
@@ -15,6 +15,10 @@ import (
 // extract all the directives definitions from the .c and .cpp files in
 // sourcePath and its subdirectories, then output the corresponding directive
 // masks map named "directives" and matchFunc named "Match" via writer.
-func Generate(sourcePath string, writer io.Writer) error {
-	return genFromSrcCode(sourcePath, "directives", "Match", writer)
+func Generate(sourcePath string, writer io.Writer, filter []string, rawOverride []string) error {
+	override, err := getOverride(rawOverride)
+	if err != nil {
+		return err
+	}
+	return genFromSrcCode(sourcePath, "directives", "Match", writer, strList2Map(filter), override)
```

Comment:
You created a special `[]string` type to collect filters, could that have been a `map[string]struct{}` to begin with?

```go
type AllowedDirectives struct {
	directiveNames map[string]struct{}
}

func (a *AllowedDirectives) Set(s string) error {
	if a.directiveNames == nil {
		a.directiveNames = make(map[string]struct{})
	}
	a.directiveNames[s] = struct{}{}
	return nil
}
```

Reply:
Fixed: add `filterFlagReader` in `cmd_util.go`